playbooks/roles/edxapp/defaults/main.yml

```diff
@@ -203,6 +203,15 @@ EDXAPP_ENABLE_AUTO_AUTH: false
 # Settings for enabling and configuring third party authorization
 EDXAPP_ENABLE_THIRD_PARTY_AUTH: true
 EDXAPP_ENABLE_OAUTH2_PROVIDER: false
+EDXAPP_THIRD_PARTY_AUTH_BACKENDS:
+  - social_core.backends.google.GoogleOAuth2
+  - social_core.backends.linkedin.LinkedinOAuth2
+  - social_core.backends.facebook.FacebookOAuth2
+  - social_core.backends.azuread.AzureADOAuth2
+  - third_party_auth.appleid.AppleIdAuth
```

Comment:
Hi @kaizoku -- this breaks juniper.3 deployments because `third_party_auth.appleid.AppleIdAuth` [doesn't exist in `opencraft-release/juniper.3`](https://github.com/open-craft/edx-platform/tree/opencraft-release/juniper.3/common/djangoapps/third_party_auth) (nor [in upstream `open-release/juniper.3`](https://github.com/edx/edx-platform/tree/open-release/juniper.3/common/djangoapps/third_party_auth), it's [only in master](https://github.com/edx/edx-platform/pull/24623/files)).

The [default settings](https://github.com/open-craft/edx-platform/blob/opencraft-release/juniper.3/lms/envs/production.py#L619-L628) should suffice if there's nothing set for `EDXAPP_THIRD_PARTY_AUTH_BACKENDS` though, so not sure why they're duplicated here?

Reply:
Ach bummer, I tried using `EDXAPP_THIRD_PARTY_AUTH_BACKENDS: !!null` to see if it would fallback to the defaults in the code, but it doesn't work: 
```
 File "/edx/app/edxapp/edx-platform/lms/envs/production.py", line 630, in <module>
AUTHENTICATION_BACKENDS = list(tmp_backends) + list(AUTHENTICATION_BACKENDS)
TypeError: 'NoneType' object is not iterable
```

We could have achieved the same effect by adding `THIRD_PARTY_AUTH_BACKENDS` to [`EDXAPP_LMS_ENV_EXTRA` and/or `EDXAPP_CMS_ENV_EXTRA`](https://github.com/open-craft/configuration/blob/opencraft-release/juniper.3/playbooks/roles//edxapp/templates/lms.yml.j2) instead of adding a new variable here. But since it's already upstreamed and we're stuck with it, could you submit an internal PR to remove this `third_party_auth.appleid.AppleIdAuth` line from `opencraft-release/juniper.3`?

Reply:
Thanks for investigating this @pomegranited. #140 has the fix.

It was pointed out to me that we could use `EDXAPP_LMS_ENV_EXTRA` unfortunately after we had opened the upstream PR, and they merged this upstream fairly quickly. Also as you found we had to duplicate the default variables since setting anything else would override the defaults.